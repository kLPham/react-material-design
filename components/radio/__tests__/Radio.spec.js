import React from 'react';
import { shallow, mount } from 'enzyme';
import Radio from '../index';

// TODO: mount has a mixture of component and rendered component. I don't want to use that as a test.
describe('<Radio />', () => {
    it('should render basic radio button', () => {
        const label = 'Radio 1';
        const wrapper = shallow(
            <Radio label={label} />,
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('div').at(1)).toHaveClassName('mdc-radio');
        expect(wrapper.find('input')).toHaveClassName('mdc-radio__native-control');
        expect(wrapper.find('input')).toHaveProp('checked');
        expect(wrapper.find('label')).toHaveText(label);
    });
    it('should render disabled radio button', () => {
        const label = 'Radio 1';
        const wrapper = mount(
            <Radio disabled label={label} />,
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('div').at(1)).toHaveClassName('mdc-radio mdc-radio--disabled');
        expect(wrapper.find('input')).toHaveClassName('mdc-radio__native-control');
        expect(wrapper.find('input')).toHaveProp('checked');
        expect(wrapper.find('input')).toHaveProp('disabled');
        expect(wrapper.find('label')).toHaveText(label);
    });
});
