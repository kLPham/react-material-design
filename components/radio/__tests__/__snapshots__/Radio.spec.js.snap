// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> should render 1`] = `
<FormField
  alignEnd={false}
>
  <div
    className="mdc-radio"
  >
    <input
      checked={false}
      className="mdc-radio__native-control"
      disabled={false}
      id="mdc-radio--2e732f05-f466-4fba-acb5-2f3be58daf75"
      onChange={[Function]}
      type="radio"
    />
    <div
      className="mdc-radio__background"
    >
      <div
        className="mdc-radio__outer-circle"
      />
      <div
        className="mdc-radio__inner-circle"
      />
    </div>
  </div>
</FormField>
`;

exports[`<Radio /> should render checked 1`] = `
<FormField
  alignEnd={false}
>
  <div
    className="mdc-radio"
  >
    <input
      checked={true}
      className="mdc-radio__native-control"
      disabled={false}
      id="mdc-radio--2e732f05-f466-4fba-acb5-2f3be58daf75"
      onChange={[Function]}
      type="radio"
    />
    <div
      className="mdc-radio__background"
    >
      <div
        className="mdc-radio__outer-circle"
      />
      <div
        className="mdc-radio__inner-circle"
      />
    </div>
  </div>
  <label
    htmlFor="mdc-radio--2e732f05-f466-4fba-acb5-2f3be58daf75"
    id="mdc-radio-label--2e732f05-f466-4fba-acb5-2f3be58daf75"
  >
    Radio 1
  </label>
</FormField>
`;

exports[`<Radio /> should render disabled 1`] = `
<FormField
  alignEnd={false}
>
  <div
    className="mdc-radio"
  >
    <input
      checked={false}
      className="mdc-radio__native-control"
      disabled={true}
      id="mdc-radio--2e732f05-f466-4fba-acb5-2f3be58daf75"
      onChange={[Function]}
      type="radio"
    />
    <div
      className="mdc-radio__background"
    >
      <div
        className="mdc-radio__outer-circle"
      />
      <div
        className="mdc-radio__inner-circle"
      />
    </div>
  </div>
  <label
    htmlFor="mdc-radio--2e732f05-f466-4fba-acb5-2f3be58daf75"
    id="mdc-radio-label--2e732f05-f466-4fba-acb5-2f3be58daf75"
  >
    Radio 1
  </label>
</FormField>
`;

exports[`<Radio /> should render with a label 1`] = `
<FormField
  alignEnd={false}
>
  <div
    className="mdc-radio"
  >
    <input
      checked={false}
      className="mdc-radio__native-control"
      disabled={false}
      id="mdc-radio--2e732f05-f466-4fba-acb5-2f3be58daf75"
      onChange={[Function]}
      type="radio"
    />
    <div
      className="mdc-radio__background"
    >
      <div
        className="mdc-radio__outer-circle"
      />
      <div
        className="mdc-radio__inner-circle"
      />
    </div>
  </div>
  <label
    htmlFor="mdc-radio--2e732f05-f466-4fba-acb5-2f3be58daf75"
    id="mdc-radio-label--2e732f05-f466-4fba-acb5-2f3be58daf75"
  >
    Radio 1
  </label>
</FormField>
`;
